File: src/main/java/com/hsmchurch/app/video/application/VideoService.java
package com.hsmchurch.app.video.application;

import com.hsmchurch.app.video.domain.QVideo;
import com.hsmchurch.app.video.domain.Video;
import com.hsmchurch.app.video.domain.VideoRepository;
import com.hsmchurch.app.video.ui.response.LikeUser;
import com.hsmchurch.app.video.ui.response.ReplyForVideo;
import com.hsmchurch.app.video.ui.response.VideoDetailResponse;
import lombok.RequiredArgsConstructor;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
@RequiredArgsConstructor
public class VideoService {

    private final VideoRepository videoRepository;
    private final VideoSupportService videoSupportService;
    private final static QVideo video = QVideo.video;

    public Page<Video> listOfVideo(final Pageable pageable) {
        return videoRepository.findAll(pageable);
    }

    public Page<Video> likeList(final Long accountId,
                                final Pageable pageable) {
        final List<Long> videoIds = videoSupportService.likeList(accountId);

        return videoRepository.findAll(video.id.in(videoIds), pageable);
    }

    public Video findById(final Long id) {
        return videoRepository.findById(id)
                .orElseThrow(() -> new RuntimeException("영상을 찾을 수 없습니다."));
    }

    public VideoDetailResponse findDetailById(final Long id) {
        final Video video = findById(id);
        final List<LikeUser> likeUsers = videoSupportService.likeUsers(id);
        final List<ReplyForVideo> replies = videoSupportService.replies(id);
        return video.toDetailResponseDto(likeUsers, replies);
    }
}

File: src/main/java/com/hsmchurch/app/video/application/VideoService.kt
package com.hsmchurch.app.video.application

import com.hsmchurch.app.video.domain.Video
import com.hsmchurch.app.video.domain.VideoRepository
import com.hsmchurch.app.video.ui.response.VideoDetailResponse
import org.springframework.data.domain.Page
import org.springframework.data.domain.PageImpl
import org.springframework.data.domain.Pageable
import org.springframework.data.repository.findByIdOrNull
import org.springframework.stereotype.Service


@Service
class VideoService constructor(
        private val videoRepository: VideoRepository,
        private val videoSupportService: VideoSupportService
) {
    private fun findById(id: Long): Video =
            videoRepository.findByIdOrNull(id) ?: throw RuntimeException("영상을 찾을 수 없습니다.")

    fun listOfVideo(pageable: Pageable): Page<Video> =
        videoRepository.findAll(pageable)

    fun likeList(accountId: Long,
                 pageable: Pageable): PageImpl<Video> {
        val videoIds = videoSupportService.likeList(accountId, pageable)
        val videos = videoRepository.findAllById(videoIds).sortedByDescending { it.filmedAt }

        return PageImpl(videos, pageable, videoIds.totalElements)
    }

    fun findDetailById(id: Long): VideoDetailResponse {
        val video = findById(id)
        val likeUsers = videoSupportService.likeUsers(id)
        val replies = videoSupportService.replies(id)
        return video.toDetailResponseDto(likeUsers, replies)
    }
}
